### PR TITLE
First iteration of Spaced Repetition

### PR DESCRIPTION
**Pull request by PukiPuki**

### CODE REVIEW

File: src/main/java/seedu/address/model/AddressBook.java [hunk 1]
```diff
@@ -145,6 +148,12 @@ public void associate(Card c, Tag t) {
         cardTag.associateCardTag(c, t);
     }
 
+    //// get list of cards for review
+    public ObservableList<Card> getTodayReviewList() {
+        return new FilteredList<Card>(cards.asObservableList(),
+                (Card c) -> c.getNextReview().before(Calendar.getInstance()));
```
Comment (jethrokuan) [inline src/main/java/seedu/address/model/AddressBook.java]: refactor into predicate, and write test

File: src/main/java/seedu/address/model/card/Card.java [hunk 2]
```diff
@@ -27,19 +28,28 @@
     private final UUID id;
     private final String front;
     private final String back;
+    private final Calendar nextReview;
+    private final int lastInterval;
```
Comment (jethrokuan) [inline src/main/java/seedu/address/model/card/Card.java]: `lastInterval` and `easingFactor` should probably go into its own class. Is `lastInterval` a fixed number?

File: src/test/java/seedu/address/model/AddressBookTest.java [hunk 3]
```diff
@@ -62,6 +64,16 @@ public void getTagList_modifyList_throwsUnsupportedOperationException() {
         addressBook.getTagList().remove(0);
     }
 
+    @Test
+    public void getTodayReviewList() {
+        AddressBook newData = getTypicalAddressBook();
+        addressBook.resetData(newData);
+        List<Card> expectedList = addressBook.getCardList();
+        List<Card> actualList = addressBook.getTodayReviewList();
+        assertEquals(expectedList, actualList);
```
Comment (jethrokuan) [inline src/test/java/seedu/address/model/AddressBookTest.java]: why are they supposed to be equal?

File: src/main/java/seedu/address/model/card/Card.java (hunk 2 above)
Comment (PukiPuki) [reply src/main/java/seedu/address/model/card/Card.java]: last interval changes

File: src/main/java/seedu/address/model/AddressBook.java (hunk 1 above)
Comment (PukiPuki) [reply src/main/java/seedu/address/model/AddressBook.java]: okay

File: src/test/java/seedu/address/model/AddressBookTest.java (hunk 3 above)
Comment (PukiPuki) [reply src/test/java/seedu/address/model/AddressBookTest.java]: all new posts added are for review immediately. Predicate should return everything.
Easy test to do for now.

Comment (jethrokuan) [reply src/test/java/seedu/address/model/AddressBookTest.java]: this is not equal all the time, so the test is invalid.

File: .gitignore [hunk 4]
```diff
@@ -18,3 +18,4 @@ classes/
 /bin/
 src/main/resources/docs/
 out/
+myTests.sh
```
Comment (jethrokuan) [inline .gitignore]: remove this change, ignore locally instead.

File: src/main/java/seedu/address/model/card/Schedule.java [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+
+/**
+ * Holds the Schedule information for a Card
+ */
+public class Schedule {
+
+    private final LocalDateTime nextReview;
+    private final int lastInterval = 1;
```
Comment (jethrokuan) [inline src/main/java/seedu/address/model/card/Schedule.java]: document in code what these refer to.

File: src/main/java/seedu/address/model/card/Schedule.java [hunk 6]
```diff
@@ -0,0 +1,30 @@
+package seedu.address.model.card;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+
+/**
+ * Holds the Schedule information for a Card
+ */
+public class Schedule {
```
Comment (jethrokuan) [inline src/main/java/seedu/address/model/card/Schedule.java]: override `equals`

File: src/main/java/seedu/address/model/card/Card.java [hunk 7]
```diff
@@ -27,19 +27,22 @@
     private final UUID id;
     private final String front;
     private final String back;
+    private final Schedule schedule;
```
Comment (jethrokuan) [inline src/main/java/seedu/address/model/card/Card.java]: either add issue to add tests, or add tests now

**Comment by jethrokuan** [comment]
I'm not going to merge this, please make your changes in different branches.

File: src/main/java/seedu/address/model/card/Card.java (hunk 7 above)
Comment (PukiPuki) [reply src/main/java/seedu/address/model/card/Card.java]: I added tests

**Pull request closed without merging**